lib/gems/pending/appliance_console/cli.rb

```diff
@@ -207,6 +218,27 @@ def set_external_db
       config.post_activation
     end
 
+    DATE_REGEXP   = /^(2[0-9]{3})-(0?[1-9]|1[0-2])-(0?[1-9]|[12][0-9]|3[01])/
+    TIME_REGEXP   = /^(0?[0-9]|1[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])/
+
+    def set_date_time
+      date, time = options[:datetime].split("T")
+      if date.validate(DATE_REGEXP) && time.validate(TIME_REGEXP)
+        say "Configuring date and time"
+        autosync = auto_sync? ? options[:autosync] : false
+        config_datetime = ApplianceConsole::DateTimeConfiguration.new(:date      => date,
+                                                                      :time      => time,
+                                                                      :auto_sync => autosync)
+        raise "Failed setting date time." unless config_datetime.activate
+      else
+        raise "Failed to configure date time. Wrong format."
+      end
+    end
+
+    def validate(regex)
+      !self[regex].nil?
+    end
```

Comment:
This feels unnecessary...for conditional purposes you don't need to explicitly check nil, so the callers are currently this:

`date.validate(DATE_REGEXP) && time.validate(TIME_REGEXP)`

but can just as easily be this without introducing a helper method:

`date =~ DATE_REGEXP && time =~ TIME_REGEXP`